regression.py

```diff
@@ -25,6 +24,56 @@ def read_info():
   info.reset_index(drop=True, inplace=True)
   return info
 
+def gic_stats(data_types=['GIC']):
+  import pickle
+  temp_pkl = CONFIG['dirs']['data'] + '\_regression\gic_stats.pkl'
+
+  if os.path.exists(temp_pkl):
+    # read gic_std and gic_max from it.
+    logger.info(f"  Reading gic_std and gic_maxabs from {temp_pkl}")
+    with open(temp_pkl, 'rb') as f:
+      gic_std, gic_maxabs = pickle.load(f)
+
+  else:
+    from swerve import site_read, site_stats, read_info_df
+
+    info_df = read_info_df(exclude_errors=True)
+    #info_df = read_info()
+    sites = info_df['site_id']
+
+    gic_std = []
+    gic_maxabs = []
+
+    data = {}
+    stats = {}
+    for i,sid in enumerate(sites):
+
+      data[sid] = site_read(sid, data_types=data_types, logger=logger)
+
+      # Extract gic_std and gic_maxabs from existing stats information in data[sid]
+      # (no need to add gic_max to stats b/c it can be derived from min/max)
+      stats[sid] = site_stats(sid, data[sid], data_types=data_types, logger=logger)
```

Comment:
I've created a bit of a problem. In main.py, site_stats modifies the cache file that site_read reads if available. So if main.py was already run, the call to site_stats is not needed b/c the stats will be in the returned data[sid]. Not sure how to deal with this yet, but as-is is awkward. For now leave it and ideally we'll think of a way to improve.